src/Amalgam/AssetManager.cpp

```diff
@@ -111,65 +117,162 @@ void AssetManager::AssetParameters::UpdateResources()
 	}
 }
 
-EvaluableNodeReference AssetManager::LoadResource(AssetParameters &asset_params, EvaluableNodeManager *enm,
+//returns a pair of success followed by version number if a version can be found in an Amalgam metadata file
+static std::pair<bool, std::string> FindVersionStringInAmlgMetadata(std::ifstream &file)
+{
+	//read 200 bytes and null terminate
+	std::array<char, 201> buffer;
+	file.read(buffer.data(), 200);
+	buffer[200] = '\0';
+	std::string str(buffer.data(), 200);
+
+	//match on semantic version
+	std::regex pattern("version (\\d+\\.\\d+\\.\\d+(?:-\\w+\\.\\d+)?(?:-\\w+)?(?:\\+\\w+)?(?:\\.\\w+)?)");
+
+	std::smatch match;
+	if(std::regex_search(str, match, pattern))
+		return std::make_pair(true, match[1].str());
+	else
+		return std::make_pair(false, "");
+}
+
+//returns a pair of success followed by version number if a version can be found in an Amalgam execute on load file
+std::pair<bool, std::string> FindVersionStringInAmlgExecOnLoad(std::ifstream &file)
+{
+	//read 200 bytes and null terminate
+	std::array<char, 201> buffer;
+	file.read(buffer.data(), 200);
+	buffer[200] = '\0';
+	std::string str(buffer.data(), 200);
```

Comment:
Should this code be split out into a helper function?  (I think the byte-level char array manipulation should be sketchy in modern C++ but can't immediately find an alternative, and it's easier to justify in comments just once.)

Reply:
(Doesn't need action now.)